### PR TITLE
Modify reset experience to make idiot proof and minimize accidental r…

### PR DESCRIPTION
**Pull request by ahickey**
Modify the reset flow / messaging to minimize accidents.

### CODE REVIEW

File: src/script/src/gesi.ts [hunk 1]
```diff
@@ -89,15 +89,29 @@ function setMainCharacter() {
...
+  if (reallySure.getSelectedButton() !== ui.Button.OK) return;
+
+  if (reallySure.getResponseText() != numChars.toString()) { ui.alert("Incorrect response received.  RESET aborted."); return; }
+
+  Object.keys(authenticatedCharacters).forEach((characterName: string) => {
     getClient(characterName).reset();
   });
 
   getDocumentProperties_().deleteAllProperties();
+  Logger.log("Reset performed successfully"); // audit this reset request
```
Comment (Blacksmoke16) [inline src/script/src/gesi.ts:113]:
```suggestion
```
Don't really need logging.

File: src/script/src/gesi.ts [hunk 2]
```diff
@@ -89,15 +89,29 @@ function setMainCharacter() {
 // @ts-ignore
 function reset() {
   const ui = SpreadsheetApp.getUi();
-  const response = ui.alert('Reset?', 'Are you sure you want to reset your data?', ui.ButtonSet.YES_NO);
 
-  if (response === ui.Button.NO) return;
+  const response = ui.alert('Reset?', 'Are you sure you want to reset ALL character data in this project?', ui.ButtonSet.YES_NO);
```
Comment (Blacksmoke16) [inline src/script/src/gesi.ts:93]:
```suggestion
  const response = ui.alert('Reset?', 'Are you sure you want to reset the authentication data of ALL characters on this sheet, including those added by other users?', ui.ButtonSet.YES_NO);
```

File: src/script/src/gesi.ts [hunk 3]
```diff
@@ -89,15 +89,29 @@ function setMainCharacter() {
...
 
-  Object.keys(getAuthenticatedCharacters()).forEach((characterName: string) => {
+  if (response !== ui.Button.YES) { ui.alert("Reset aborted"); return; }
```
Comment (Blacksmoke16) [inline src/script/src/gesi.ts:95]:
```suggestion
  if (response !== ui.Button.YES) { return; }
```
Given they either click away or `no`, don't think we need to do another alert.

**Review by Blacksmoke16 — COMMENTED** [review]
Overall looks good to me! Thanks for this. Left a few minor comments then is good to go. :+1:.

**Comment by ahickey** [comment]
Updated with comments

**Review by Blacksmoke16 — APPROVED** [review]
Thanks!

**Pull request merged**